### ISSUE TITLE
`ocrd-cis-align --help` and other commands fail to show help

### ISSUE THREAD

**Issue by stweil** [cisocrgroup/ocrd_cis#39]
`ocrd-cis-align` does not support the standard `--help` option, but fails with a `KeyError` exception.

**Comment by stweil** [comment]
Other commands which also don't support `--help`: ocrd-cis-data.

**Comment by finkf** [comment]
I thought, that the ocrd core tools take care of the standard command line options.  I will look into it.

**Comment by stweil** [comment]
Yes, they provide at least most of the required functionality. CC'ing @kba who might be able to give more information.

**Comment by finkf** [comment]
I just installed the latest dev version using venv and it seems to work for me:
```
$ ocrd-cis-align --help
Usage: ocrd-cis-align [OPTIONS]

Options:
  -V, --version                   Show version
  -l, --log-level [OFF|ERROR|WARN|INFO|DEBUG|TRACE]
                                  Log level
  -J, --dump-json                 Dump tool description as JSON and exit
  -p, --parameter TEXT            Parameters, either JSON string or path to
                                  JSON file
  -g, --page-id TEXT              ID(s) of the pages to process
  -O, --output-file-grp TEXT      File group(s) used as output.
  -I, --input-file-grp TEXT       File group(s) used as input.
  -w, --working-dir TEXT          Working Directory
  -m, --mets TEXT                 METS URL to validate
  --help                          Show this message and exit.
```

**Comment by stweil** [comment]
Hm. I just tested an installation with the latest `OCR-D/ocrd_all`, and I still get the problem.

**Comment by stweil** [comment]
The problem can also be reproduced with the latest code from the `dev` branch on Debian testing:

    cd ocrd_cis/
    python3 -m venv venv
    source venv/bin/activate
    pip install -U pip
    pip install .
    ocrd-cis-align --help

=>

    Traceback (most recent call last):
      File "/ocrd_cis/venv/bin/ocrd-cis-align", line 11, in <module>
        load_entry_point('ocrd-cis==0.0.6', 'console_scripts', 'ocrd-cis-align')()
    [...]
        ocrd_tool['input_file_grp'],
    KeyError: 'input_file_grp'

**Comment by kba** [comment]
All processors need to explicitly define `input_file_grp` and `output_file_grp` in the ocrd-tool.json currently. However for ocrd_cis:

```xml
ocrd validate tool-json ocrd-tool.json |grep input
  <error>[tools.ocrd-cis-ocropy-rec] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-ocropy-train] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-align] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-wer] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-jar] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-profile] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-train] 'input_file_grp' is a required property</error>
  <error>[tools.ocrd-cis-post-correct] 'input_file_grp' is a required property</error>
```

@finkf Can you make sure that the ocrd-tool.json validates?

I will make the help generation in core more flexible nonetheless.

**Comment by finkf** [comment]
I am trying to validate ocrd-tool.json, but it keeps failing (see below). What is the correct way to validate the `ocrd-tool.json`?

```bash
$ ocrd ocrd-tool validate ocrd-tool.json
Usage: ocrd ocrd-tool [OPTIONS] JSON_FILE COMMAND [ARGS]...
Try "ocrd ocrd-tool --help" for help.

Error: No such command "ocrd-tool.json".
$ ocrd ocrd-tool list-tools
Usage: ocrd ocrd-tool [OPTIONS] JSON_FILE COMMAND [ARGS]...
Try "ocrd ocrd-tool --help" for help.

Error: Missing command.
$ ocrd_cis ocrd ocrd-tool list-tools ocrd-tool.json
Usage: ocrd ocrd-tool [OPTIONS] JSON_FILE COMMAND [ARGS]...
Try "ocrd ocrd-tool --help" for help.

Error: No such command "ocrd-tool.json".
$ ocrd ocrd-tool list-tools ocrd-tool.json ocrd-cis-align
Usage: ocrd ocrd-tool [OPTIONS] JSON_FILE COMMAND [ARGS]...
Try "ocrd ocrd-tool --help" for help.

Error: No such command "ocrd-tool.json".
```

**Comment by kba** [comment]
Yeah, `ocrd-tool` subcommand is confusing, the syntax is `ocrd ocrd-tool /path/to/my/ocrd-tool.json validate`. But I'm aware how weird and unintuitive that is. Therefore we also expose it as `ocrd validate tool-json`:

```sh
ocrd validate tool-json ocrd-tool.json 
```

or just

```sh
ocrd validate tool-json
```

if you're in the same directory as the ocrd-tool.json.

**Comment by finkf** [comment]
Ah. OK now it works. Thanks! I will make sure that the cis tools validate.

**Comment by finkf** [comment]
fixed in 34201b6d00216d650b3d8759c1891dc6d9d1ce64
